### ISSUE TITLE
Using CommonAppDataFolder in project.AddRegValue(RegValue) generates wrong XML

### ISSUE THREAD

**Issue by xmarshal** [oleg-shilo/wixsharp#774]
```cs
project.AddRegValue(new RegValue(RegistryHive.LocalMachine, "path", "name", "[CommonAppDataFolder]"));
```

generates:
```xml
<Component Id="Registry.3" Guid="af599c42-a2e5-4251-b7ee-49254e4671a1">
          <RegistryKey Root="HKLM" Key="path">
            <RegistryValue Id="name" Type="expandable" KeyPath="yes" Value="CommonAppDataFolder" Name="update_dir" />
          </RegistryKey>

          <CreateFolder />
</Component> 
```
expected result:
```xml
<Component Id="Registry.3" Guid="af599c42-a2e5-4251-b7ee-49254e4671a1">
          <RegistryKey Root="HKLM" Key="path">
            <RegistryValue Id="name" Type="expandable" KeyPath="yes" Value="[CommonAppDataFolder]" Name="update_dir" />
          </RegistryKey>

          <CreateFolder />
</Component>
```

So in registry value CommonAppDataFolder instead of C:\ProgramData\
If I use @"[%ProgramData]\" in registry C:\ProgramData\

If I replace "CommonAppDataFolder" with "[CommonAppDataFolder]" WixSourceGenerated event in registry "C:\ProgramData\"

**Comment by oleg-shilo** [comment]
It's a defect. Marking the issue correspondingly.

**Comment by oleg-shilo** [comment]
I found that expanding Wix constants for `RegValue` was implemented practically from the first release. At least from the first commit to GitHub.

To be honest I think it was a case of over-engineering as there is no plausible use-case for it. 

I have removed the offending routine but since it has been with WixSharp almost forever I am concerned that it may affect some users who willingly or unwillingly became dependent on the expanding routine. Thus by default, WixSharp will not expand WiX constants present in Reg value.

Though if for whatever reason you want to restore the old behavier it can be done as follow:
```C#
AutoElements.ExpandWixEnvInRegValue = true;
```
